### ISSUE TITLE
Uncertain error without dockerfile directive

### ISSUE THREAD

**Issue by nabokihms** [werf/werf#3036]
Hello! I encountered an uncertain error when I accidentally wrote `context` instead of `dockerfile` in a werf.yaml.

werf.yaml
```yaml
configVersion: 1
project: dashboard
---
image: frontend
context: Dockerfile.frontend
```

Logs:
```
Version: v1.2.2+fix1
┌ Running GC for tmp data
│ /private/var/folders/wd/5jrrjzpd6dzb75n7sg8nfg480000gn/T/werf-project-data-354066578
│ /private/var/folders/wd/5jrrjzpd6dzb75n7sg8nfg480000gn/T/werf-project-data-758214449
│ /Users/m.nabokikh/.werf/service/tmp/released/projects/werf-project-data-354066578
│ /Users/m.nabokikh/.werf/service/tmp/released/projects/werf-project-data-758214449
│ /private/var/folders/wd/5jrrjzpd6dzb75n7sg8nfg480000gn/T/werf-config-render-319227034
│ /private/var/folders/wd/5jrrjzpd6dzb75n7sg8nfg480000gn/T/werf-config-render-863334975
│ /Users/m.nabokikh/.werf/service/tmp/created/werf_config_renders/werf-config-render-319227034
│ /Users/m.nabokikh/.werf/service/tmp/created/werf_config_renders/werf-config-render-863334975
└ Running GC for tmp data (0.00 seconds) FAILED

Running time 0.04 seconds
Error: unable to load werf config: tmp manager GC failed: unable to remove tmp projects dirs /private/var/folders/wd/5jrrjzpd6dzb75n7sg8nfg480000gn/T/werf-project-data-354066578, /private/var/folders/wd/5jrrjzpd6dzb75n7sg8nfg480000gn/T/werf-project-data-758214449: Status: , Code: 125
```

However, it would be better to see something like:
```
Something wrong with the configuration for building image "frontend".
```

**Comment by distorhead** [comment]
This error does not relate to the dockerfile-image configuration. It is `Running GC for tmp data (0.00 seconds) FAILED` process that failed during build command execution. GC for tmp data removes some tmp data that has been generated during previous builds.

Seems like werf does not have a permission to remove `/tmp` files.

**Comment by distorhead** [comment]
I think it is safe to transform GC error to warning and continue build process in such case.

**Comment by nabokihms** [comment]
This is my very first build, and after changing context to dockerfile everything works fine. May be there is something missing in the "quick start" guide.

**Comment by distorhead** [comment]
> Seems like werf does not have a permission to remove /tmp files.

@nabokihms Could you check this?

**Comment by nabokihms** [comment]
Yes, it looks like werf can do anything it wants with the tmp directory.

**Comment by distorhead** [comment]
1. Now werf will display the following error for your configuration:

![image](https://user-images.githubusercontent.com/1161992/102334593-e9cfed00-3f9f-11eb-8a86-4b6e3109ba3c.png)

 2. The original error in the issue related to the permissions of GC process. Werf can run GC process at the beginning of any main command (like werf-build, or werf-converge, or werf-dismiss) when there is too much tmp data accumulated. 
 
Maybe this directory `/private/var/folders/wd/5jrrjzpd6dzb75n7sg8nfg480000gn/T/werf-project-data-354066578` contains some files owned by the root user or another user. It is 99% macos-related, we have encountered this error on our test macos: https://github.com/werf/werf/runs/1559895347?check_suite_focus=true

So I will rename this issue.

**Comment by alexey-igrychev** [comment]
Both parts of this are resolved: the misleading config error was fixed shortly after the report, and tmp-data GC no longer aborts the command when it can't remove a path (path-removal errors are tolerated since v2.44.0). Closing as resolved.
